execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1798,4 +1803,54 @@ where
             commit_result => Ok(commit_result),
         }
     }
+
+    /// Obtains validator weights for given era.
+    pub fn get_era_validators(
+        &self,
+        correlation_id: CorrelationId,
+        get_era_validators_request: GetEraValidatorsRequest,
+    ) -> Result<GetEraValidatorsResult, Error> {
```

Comment:
To elaborate on the comment about `GetEraValidatorsResult` - if it was `GetEraValidatorsError`, it could have a variant like `Other(Error)`.

It seems a bit weird to receive something like `Ok(RootNotFound)`, as it seems more fit to be an error rather than a value indicating success. But maybe it makes sense in some wider context I'm not seeing right now?

Reply:
Done